src/MongoDB.Driver.Legacy/Builders/IndexKeysBuilder.cs

```diff
@@ -158,6 +168,34 @@ public IndexKeysBuilder Ascending(params string[] names)
             return this;
         }
 
+        /// <summary>
+        /// Sets wildcard keys to index in ascending order. If the wildcard names collection is empty, the generated key will be for the whole collection.
+        /// </summary>
+        /// <param name="names">The wildcard keys.</param>
+        /// <returns>The builder (so method calls can be chained).</returns>
+        public IndexKeysBuilder AscendingWithWildcard(params string[] names)
+        {
+            const string wildcardTemplate = "$**";
+
+            if (names?.Any() != true)
+            {
+                _document.Add(wildcardTemplate, 1);
+            }
+            else
+            {
+                foreach (var name in names)
+                {
```

Comment:
Not sure which one is better here. My version looks more clear/simple to me.

Reply:
Note: Maybe the below can be a problem:
As I mention [here](https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/28#discussion_r278873889) the `AscendingWithWildcard` works in another way with just `Ascending`. In case if `Ascending` has empty input parameters, this method does nothing implicitly. But `AscendingWithWildcard` will generate `$**"` for the empty parameters. I guess it may be named as logic `inconsistency`. What do you think?

Reply:
> Not sure which one is better here. My version looks more clear/simple to me.

Keeping your version is fine with me :) Just wanted to suggest a possible alternative!

Reply:
> Note: Maybe the below can be a problem:
> As I mentioned [here](https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/28#discussion_r278873889) the `AscendingWithWildcard` works in another way with just `Ascending`. In case if `Ascending` has empty input parameters, this method does nothing implicitly. But `AscendingWithWildcard` will generate `$**"` for the empty parameters. I guess it may be named as logic `inconsistency`. What do you think?

If we do choose to update the legacy API, it seems reasonable to keep the logic as similar to the previous `Ascending` function which did nothing when passed empty input parameters. If the user truly wants `$**: 1` they can pass `$**` in as one of the parameters.

Reply:
My original idea was to hide the `mongo query syntax` under this method at least if it's possible. But in your case, it will not work :( 
I guess the original behavior is not really correct here when we allowed having the empty parameter for `Ascending` method. It doesn't make sense(?). If so, should we take it into account when we develop new features?

Reply:
Hm, good point regarding hiding the mongo query syntax. I'm unsure how common a case passing in zero parameters/an empty array is. Perhaps the behavior here for the new method doesn't matter as much as long as we're consistent with the other Legacy API functions? Deferring to @rstam for this one :)

Reply:
I made a mistake during the implementation. These methods should not be called `Ascending` since it's not their business logic. If I change it, the question about `inconsistency` will disappear. I will do it in next commit.